autogpt/agent/agent.py

```diff
@@ -252,11 +253,27 @@ def start_interaction_loop(self):
                     result = plugin.post_command(command_name, result)
                 if self.next_action_count > 0:
                     self.next_action_count -= 1
-            memory_to_add = (
-                f"Assistant Reply: {assistant_reply} "
-                f"\nResult: {result} "
-                f"\nHuman Feedback: {user_input} "
-            )
+                    
+            if command_name != "do_nothing":
+                if isinstance(self.memory, PineconeMemory):
+                    stats = self.memory.get_stats()
+                    index_fullness = stats.index_fullness
+                    if index_fullness == 1:
+                        should_clear = clean_input(
+                            f"""Your Pinecone index is {Fore.RED}100% full{Style.RESET_ALL}.
+                                No new data can be stored.
+                                Clear index? (y/n):"""
+                        )
+                        if should_clear.lower() == "y":
+                            self.memory.clear
+                        else:
+                            pass
```

Comment:
we should abstract this out a bit more to be compatible with any memory backend without a huge chain of if statements

Reply:
This should probably live in the memory base class